c3/optimizers/optimizer.py

```diff
@@ -251,10 +251,10 @@ def fct_to_min(
         [np.ndarray, tf.constant]
             Value of the goal function. Float if input is np.array else tf.constant
         """
-        self.optim_status["params"] = [
-            par.get_other_value(y)
-            for par, y in zip(self.pmap.get_parameters(), np.array(input_parameters))
-        ]
+        pars = []
+        for par, y in zip(self.pmap.get_parameters(), np.array(input_parameters)):
```

Comment:
`zip` is a nifty tool that helps write very concise code but sometimes makes it difficult to understand the programmer's intent when read at a later stage. Can you add a comment on what this `zip` is doing here?